src/rules/max-scenarios-per-file.js

```diff
@@ -0,0 +1,40 @@
+var rule = 'max-scenarios-per-file';
+
+function maxScenariosPerFile(feature, _, config) {
+  var errors = [];
+  var count = 0;
+  var maxScenarios = config;
+
+  if (feature.children) {
+    count = count + feature.children.length;
+
+    feature.children.forEach(function (scenario) {
+      if (scenario.type === 'Background') {
+        count = count - 1;
+      }
+
+      if (scenario.examples) {
+        count = count - 1;
+        scenario.examples.forEach(function (example) {
+          count = count + example.tableBody.length;
+        });
+      }
+    });
+  }
+
+  if (count > maxScenarios) {
+    errors.push({
+      message: 'Number of scenarios exceeds maximum: ' + count + '/' + maxScenarios,
+      rule,
+      line: 0
+    });
+  }
+
+  return errors;
+}
+
+module.exports = {
+  name: rule,
+  run: maxScenariosPerFile,
+  availableConfigs: 10
```

Comment:
The way you're trying to set up the rule configuration will not work. 

The rule configuration can be one of these 3 categories:
1. Rule config is a string that can be "on" or "off" (eg `"no-superfluous-tags": "on")
2. Rule config is a 2 element array where the first element is "on" or "off" and the second element is another string (eg. `"new-line-at-eof": ["on", "yes"]`). In that case all the valid values for the second string, need to be exported by the rule, as "availableConfigs". 
3. Rule config is a 2 element array where the first element is "on" or "off" and the second element is a map. All the valid keys for the map are exported by the `availableConfigs` of the rule. Example: https://github.com/vsiakka/gherkin-lint/blob/master/src/rules/no-restricted-tags.js#L3

The code that does the validation I mentioned above lives here: https://github.com/vsiakka/gherkin-lint/blob/master/src/config-verifier.js#L15

You're trying to use the 2nd type of configuration, but because any numerical value should be valid as a config  you can't really make an exhaustive list of valid config values, so you should use the 3rd type of configuration, and use a map as an input eg:
```
avaliableConfigs = {
 maxSenarior: {defaultValue}
```

Reply:
Thanks, done the update to it.